pkg/graveler/settings/manager_test.go

```diff
@@ -301,17 +253,13 @@ func prepareTest(t *testing.T, ctx context.Context, kvEnabled bool, cache cache.
 		cb = branchLockCallback
 	}
 	var opts []settings.ManagerOption
-	if cache != nil {
-		opts = append(opts, settings.WithCache(cache))
+	if cache == nil {
+		cache = &nopCache{}
```

Comment:
Just to make sure this is the new behaviour we like:  
Not the same logic - the previous implementation left NewManager with default settings unless there was 'cache' reference passed.
The caller cloud pass the nopCache.
This new implementation will always set case based on the input - in this case I would move the nopCache nil check inside 'settings.WithCache'

Reply:
Look at the way the manager is initialized - It creates its own cache. Should we change the behaviour to initialize the cache from the caller side?